breadbox/breadbox/celery_task/utils.py

```diff
@@ -204,3 +204,25 @@ def update_state(
         meta["message"] = message
 
     task.update_state(state=state, meta=meta)
+
+
+def check_celery():
+    """
+    Checks to see if celery redis broker is connected.
+    Pings to see if any worker responds and returns true if successful response detected
+    """
+    inspect = app.control.inspect()
+    try:
```

Comment:
I don't know that we need to explicitly connect to the broker before submitting a task. If the broker is offline, then we already get an exception thrown when it tries to submit the task to the broker. 

That being said, I suppose it's harmless to try. Probably just costs a few milliseconds and I can't imagine it having any real downside.

Reply:
@pgm  Usually if celery isn't running (at least on development) we've forgotten to start the celery broker. I figured returning an explicit error to the client could be helpful.. This does have 3 retries to connect to the broker which takes a few seconds rather than a few milliseconds though so maybe I should decrease the amount of retries